### PR TITLE
Prevent attempts to load null values using the cursor.getFoo(index) meth...

### PR DESCRIPTION
**Pull request by mcumings**
Prevent attempts to load null values using the cursor.getFoo(index) method calls on nullable, non-primitive fields.

The new behavior is to detect nullable field values and wrap their loading behavior in a Cursor.isNull(index) check.  This allows for a distinction between values which are not present and those which have their default (Java) value.

Also fixed a potential test escape and added more granular resolution to BoxedValuesTest.

### CODE REVIEW

**Comment by mcumings** [comment]
Not sure how you're managing develop versus master.  Is this change better suited to a merge to master?

File: library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedModel.java [hunk 1]
```diff
@@ -6,26 +6,63 @@
...
     @Column(notNull = true)
-    public int sort;
+    public short shortPrimitiveFieldNotNull = 1;
 
-    @Column
-    public Integer rblNumber;
+    @Column(notNull = true)
+    public Short shortFieldNotNull = 1;
+
+    @Column()
```
Comment (agrosner) [inline library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedModel.java]: remove paranthesis

File: library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedModel.java [hunk 2]
```diff
@@ -6,26 +6,63 @@
...
+    public Short shortField = 1;
+
+    @Column(notNull = true)
+    public int integerPrimitiveFieldNotNull = 1;
+
+    @Column(notNull = true)
+    public Integer integerFieldNotNull = 1;
+
+    @Column()
```
Comment (agrosner) [inline library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedModel.java]: same as above

File: library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedModel.java [hunk 3]
```diff
@@ -6,26 +6,63 @@
...
+    public Integer integerField = 1;
+
+    @Column(notNull = true)
+    public long longPrimitiveFieldNotNull = 1L;
+
+    @Column(notNull = true)
+    public Long longFieldNotNull = 1L;
+
+    @Column()
```
Comment (agrosner) [inline library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedModel.java]: same as above

File: library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedModel.java [hunk 4]
```diff
@@ -6,26 +6,63 @@
...
+    public Long longField = 1L;
+
+    @Column(notNull = true)
+    public float floatPrimitiveFieldNotNull = 1.0f;
+
+    @Column(notNull = true)
+    public Float floatFieldNotNull = 1.0f;
+
+    @Column()
```
Comment (agrosner) [inline library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedModel.java]: same as above

File: library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedModel.java [hunk 5]
```diff
@@ -6,26 +6,63 @@
...
 
     @Column(notNull = true)
-    public int area;
+    public double doublePrimitiveFieldNotNull = 1.0;
 
     @Column(notNull = true)
-    public String station;
+    public Double doubleFieldNotNull = 1.0;
+
+    @Column()
```
Comment (agrosner) [inline library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedModel.java]: same as above

File: library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedModel.java [hunk 6]
```diff
@@ -6,26 +6,63 @@
...
+    public Double doubleField = 1.0;
 
     @Column(notNull = true)
-    public String stand;
+    public String stringFieldNotNull = "1";
+
+    @Column()
```
Comment (agrosner) [inline library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedModel.java]: same as above.

File: library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedValueTest.java [hunk 7]
```diff
@@ -3,24 +3,129 @@
...
+    private BoxedModel testObject;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        testObject = new BoxedModel();
+        testObject.id = SEQUENCE_ID.getAndIncrement();
+    }
+
+    public void testBoxedValues_nullId() {
```
Comment (agrosner) [inline library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedValueTest.java]: dont believe we need to test every field here to see if its saveable or not. rather all we need to test is a non-null and normal field can be saved.

File: compiler/src/main/java/com/raizlabs/android/dbflow/processor/utils/ModelUtils.java [hunk 8]
```diff
@@ -137,21 +138,36 @@ public static void writeLoadFromCursorDefinitionField(JavaWriter javaWriter, Pro
         }
 
         javaWriter.emitStatement(getColumnIndex(columnName));
-        javaWriter.beginControlFlow("if (%1s != -1) ", "index" + columnName);
+        String index = "index" + columnName;
+        javaWriter.beginControlFlow("if (%1s != -1) ", index);
 
         String cursorStatment = ModelUtils.getCursorStatement(newFieldType, columnName);
+        AdapterQueryBuilder nullableQueryBuilder = new AdapterQueryBuilder(queryBuilder.getQuery());
```
Comment (agrosner) [inline compiler/src/main/java/com/raizlabs/android/dbflow/processor/utils/ModelUtils.java]: can you write this without using a second ```AdapterQueryBuilder```?

**Comment by agrosner** [comment]
@mcumings looks pretty good. Just a few comments, but also this PR is not mergeable. Please fix! But yes all PRs should go against develop.

**Comment by mcumings** [comment]
I had added all the individual cases to BoxedValueTest to dd coverage around com.raizlabs.android.dbflow.processor.writer.LoadCursorWriter#CURSOR_METHOD_MAP but have removed them as requested.

Also updated the ModelUtils logic to extract the assignment logic into a separate method containing a single builder.  I like this approach better anyhow so thanks for that. :)

**Comment by agrosner** [comment]
@mcumings thanks for your help! I am always happy to accept pull requests.

**Comment by mcumings** [comment]
No sweat, good sir.

**Comment by agrosner** [comment]
@mcumings adding you as a contributor at the bottom of the README!
